routing/pathfind.go

```diff
@@ -141,7 +143,40 @@ func newRoute(sourceVertex route.Vertex,
 			amtToForward     lnwire.MilliSatoshi
 			fee              lnwire.MilliSatoshi
 			outgoingTimeLock uint32
+			tlvPayload       bool
 		)
+
+		// Define a closure that determines whether or not a feature
+		// vector properly supports a given feature.
+		supports := func(fv *lnwire.FeatureVector,
+			bit lnwire.FeatureBit) bool {
+
+			if fv == nil {
+				log.Debugf("Node %v has no features",
+					edge.Node.PubKeyBytes)
+				return false
+			}
+
+			err := feature.ValidateDep(fv, bit)
```

Comment:
Is it still necessary to validate at this point? I had the impression we were now checking all these features 'at the gate'?

Reply:
currently only the invoice feature bits are checked, but not the ones on the node announcement. we could move to only do this validation there instead of for the final hop as well